src/a_star.cpp

```diff
@@ -253,9 +255,9 @@ void AStarAlgorithm<NodeT>::setGoal(
     if (!_start) {
       throw std::runtime_error("Start must be set before goal.");
     }
-
+    const std::shared_ptr<costmap_2d::Costmap2DROS> costmap_2d_ros = _collision_checker->getCostmapROS();
```

Comment:
Q: can we use const here?
```suggestion
    const std::shared_ptr<const costmap_2d::Costmap2DROS> costmap_2d_ros = _collision_checker->getCostmapROS();
```

Reply:
This change is not required, I changed something here and then forgot to revert this. 
But we can't change that to const because resetObstacleHeuristic doesnot accept that as parameter